packages/runtime-config-provider/src/blob-reader.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { RestError } from '@azure/ms-rest-js';
+import { BlobClient, BlobServiceClient } from '@azure/storage-blob';
+import { inject, injectable } from 'inversify';
+import { isNil } from 'lodash';
+import { BlobServiceClientFactory, runtimeConfigIocTypes } from './ioc-types';
+
+export interface BlobContentUpdateResponse {
+    isModified: boolean;
+    notFound: boolean;
+    updatedContent: string;
+}
+
+@injectable()
+export class BlobReader {
+    private readonly blobServiceClient: BlobServiceClient;
+
+    constructor(@inject(runtimeConfigIocTypes.BlobServiceClientFactory) blobServiceClientProvider: BlobServiceClientFactory) {
+        this.blobServiceClient = blobServiceClientProvider();
```

Comment:
Would be good if you do something like blobServiceClientProvider.createBlogServiceClient(), it is more readable in my opinion.